tailcfg/tailcfg.go

```diff
@@ -1369,3 +1373,59 @@ type SetDNSRequest struct {
 	// Value is the value to add.
 	Value string
 }
+
+// SSHPolicy is the policy for how to handle incoming SSH connections
+// over Tailscale.
+type SSHPolicy struct {
+	// Rules are the rules to process for an incoming SSH
+	// connection. The first matching rule takes its action and
```

Comment:
Does this mean that if I have multiple rules, they're evaluated in order? I guess that makes sense. If we're allowing you to specify multiple SSH users though, and want to default to the non-root one, but you list them in a different order, does that mean we might use the root one instead?

Reply:
Yeah, but keep in mind that this SSHPolicy is not the user facing bits at all. This is the internal representation that tailscaled can handle, analogous to how ACL JSON compiles down to the internal PacketFilter/FilterRule:

https://pkg.go.dev/tailscale.com/tailcfg#MapResponse.PacketFilter
https://pkg.go.dev/tailscale.com/tailcfg#FilterRule

So this is just setting up tailscaled to be able to do configurable sets of things from control where we can implement higher level policies and UIs that map down to this.